File: typescript/parser/src/parser/json/type/jbool.ts
import { JValue } from './jvalue'

export class JBool extends JValue {
	value: boolean;

	constructor(value: boolean) {
		super()
		this.value = value
	}

	equals(other: any) {
		return other instanceof JBool && this.value == other.value
	}
}

File: scala/parsers/src/main/scala/jp/co/teruuu/parser/json/JBool.scala
package jp.co.teruuu.parser.json

case class JBool(value: Boolean) extends JValue {
  override def equals(obj: Any): Boolean = obj.isInstanceOf[JBool] && value.equals(obj.asInstanceOf[JBool].value)

  def toSpray(): spray.json.JsValue = {
    spray.json.JsBoolean(value)
  }
}
